collective/auditlog/utils.py

```diff
@@ -18,17 +18,38 @@
 from zope.globalrequest import getRequest
 
 
+def getSite():
+    site = getSiteHook()
+    if site is None:
+        # user might be at zope root level. Try to guess site
+        zope_root = app()
+        plone_sites = zope_root.objectIds("Plone Site")
+        if len(plone_sites) == 1:
+            # just one plone site, safe bet
+            site = zope_root[plone_sites[0]]
+        elif len(plone_sites) > 1:
+            # many sites. Might be an undo attempt
+            request = getRequest()
+            if request and "transaction_info" in request.other:
+                info = " ".join(request.other["transaction_info"])
+                for plone_site in plone_sites:
+                    if " /{}/".format(plone_site) in info:
+                        site = zope_root[plone_site]
+    return site
+
+
 def is_installed():
     try:
         site = getSite()
         installer = get_installer(site)
         installed = installer.is_product_installed("collective.auditlog")
         registry = queryUtility(IRegistry, context=site)
-        installed = (
-            installed
-            and registry
-            and ("collective.auditlog.interfaces.IAuditLogSettings.storage" in registry)
+        storage = (
+            registry.get("collective.auditlog.interfaces.IAuditLogSettings.storage")
+            if registry
+            else False
         )
+        installed = installed and storage
```

Comment:
I know this is not what you patch correctly fixes (a missing registry) but is_installed should check for the existence of `getEngine()` because the configuration can be also setup through the zope.conf since the stuff from #26 was merged on master

Reply:
Done. The signature of ``getSession`` allows to pass connection parameters too - but it is not used in collective.auditlog. If a session would pass custom connection paramters wether the engine would be created in a wrong way (if there is also a connection parameter reg/config entry) or ``is_installed`` would fail.